test/txsim/run_test.go

```diff
@@ -152,3 +157,50 @@ func Setup(t testing.TB) (keyring.Keyring, string, string) {
 
 	return cctx.Keyring, rpcAddr, grpcAddr
 }
+
+func TestTxSimUpgrade(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping TestTxSimUpgrade in short mode.")
+	}
+	cp := app.DefaultConsensusParams()
+	cp.Version.AppVersion = v2.Version
+	cfg := testnode.DefaultConfig().
+		WithTimeoutCommit(300 * time.Millisecond).
+		WithConsensusParams(cp).
+		WithFundedAccounts("txsim-master")
+	cctx, _, grpcAddr := testnode.NewNetwork(t, cfg)
+
+	// updrade to v3 at height 20
+	sequences := []txsim.Sequence{
+		txsim.NewUpgradeSequence(v3.Version, 20),
+	}
+
+	opts := txsim.DefaultOptions().
+		// SuppressLogs().
+		WithPollTime(time.Millisecond * 100)
+
+	err := txsim.Run(
+		cctx.GoContext(),
+		grpcAddr,
+		cctx.Keyring,
+		encoding.MakeConfig(app.ModuleEncodingRegisters...),
+		opts,
+		sequences...,
+	)
+	require.NoError(t, err)
+
+	conn, err := grpc.NewClient(grpcAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	require.NoError(t, err)
+	defer conn.Close()
+
+	querier := signaltypes.NewQueryClient(conn)
+
+	// We can't check that the upgrade was successful because the upgrade height is thousands of blocks away
+	// and even at 300 millisecond block times, it would take too long. Instead we just want to assert
+	// that the upgrade is ready to be performed
```

Comment:
👍 good idea. As a separate issue we can explore making it easy to override the upgrade delay so that tests can configure it to be 1 block and verify the upgrade executed and app version transition from 2 -> 3.

Reply:
I'm working on that right now actually